src/server/pkg/storage/fileset/writer.go

```diff
@@ -68,13 +69,15 @@ func newWriter(ctx context.Context, store Store, tracker track.Tracker, chunks *
 	if w.noUpload {
 		chunkWriterOpts = append(chunkWriterOpts, chunk.WithNoUpload())
 	}
-	w.additive = index.NewWriter(ctx, chunks, "additive-index-writer-"+uuidStr)
-	w.deletive = index.NewWriter(ctx, chunks, "deletive-index-writer-"+uuidStr)
-	w.cw = chunks.NewWriter(ctx, "chunk-writer-"+uuidStr, w.callback, chunkWriterOpts...)
+	w.additive = index.NewWriter(ctx, chunks, "additive-index-writer")
+	w.deletive = index.NewWriter(ctx, chunks, "deletive-index-writer")
+	w.cw = chunks.NewWriter(ctx, "chunk-writer", w.callback, chunkWriterOpts...)
 	return w
 }
 
 // Append creates an append operation for a file and provides a scoped file writer.
+// TODO: providing the tag here, since it is required, is a better API than getting a *FileWriter
```

Comment:
We need to be able to support multiple tags for a single file, so just associating a single tag with a file writer would not work. I do agree that this API could probably be improved, and I am leaning towards inverting the control by taking a reader instead of implementing the io.Writer interface which would allow us to provide a function that takes a tag and the reader for the tagged data.

Reply:
I will change the comment to mention taking a tag + io.Reader